sktime/forecasting/hf_transformers_forecaster.py

```diff
@@ -448,42 +572,106 @@ def get_test_params(cls, parameter_set="default"):
+                    (1, len(X), len(X.columns))
+                )
+                if X is not None
+                else X
+            )
+        else:
+            self.y = _frame2numpy(y)
+            self.X = _frame2numpy(X) if X is not None else X
+
+        self._num, self._len, _ = self.y.shape
         self.fh = fh
+        self.seq_len = seq_len
+        self.batch_size = batch_size
+        self.no_size1_batch = no_size1_batch
 
     def __len__(self):
         """Return length of dataset."""
-        return max(len(self.y) - self.seq_len - self.fh + 1, 0)
+        true_length = self._num * max(self._len - self.seq_len - self.fh + 1, 0)
+        if self.no_size1_batch and true_length % self.batch_size == 1:
+            return true_length - 1
+        else:
+            return true_length
 
     def __getitem__(self, i):
         """Return data point."""
-        from torch import from_numpy, tensor
-
-        hist_y = tensor(self.y[i : i + self.seq_len]).float()
+        from torch import tensor
+
+        m = i % (self._len - self.seq_len - self.fh + 1)
+        n = int((i - m) / self._len)
+        hist_y = tensor(self.y[n, m : m + self.seq_len, :]).float().flatten()
+        futu_y = (
+            tensor(self.y[n, m + self.seq_len : m + self.seq_len + self.fh, :])
+            .float()
+            .flatten()
+        )
         if self.X is not None:
             exog_data = tensor(
-                self.X[i + self.seq_len : i + self.seq_len + self.fh]
+                self.X[n, m + self.seq_len : m + self.seq_len + self.fh, :]
             ).float()
-            hist_exog = tensor(self.X[i : i + self.seq_len]).float()
+            hist_exog = tensor(self.X[n, m : m + self.seq_len, :]).float()
         else:
             exog_data = tensor([[]] * self.fh)
             hist_exog = tensor([[]] * self.seq_len)
+
         return {
             "past_values": hist_y,
             "past_time_features": hist_exog,
             "future_time_features": exog_data,
             "past_observed_mask": (~hist_y.isnan()).to(int),
-            "future_values": from_numpy(
-                self.y[i + self.seq_len : i + self.seq_len + self.fh]
-            ).float(),
+            "future_values": futu_y,
         }
+
+
+def _same_index(data):
+    data = data.groupby(level=list(range(len(data.index.levels) - 1))).apply(
+        lambda x: x.index.get_level_values(-1)
+    )
+    assert data.map(
+        lambda x: x.equals(data.iloc[0])
+    ).all(), "All series must has the same index"
+    return data.iloc[0], len(data.iloc[0])
+
+
+def _frame2numpy(data):
+    idx, length = _same_index(data)
+    arr = np.array(data.values, dtype=np.float32).reshape(
+        (-1, length, len(data.columns))
+    )
+    return arr
+
+
+def _to_multiindex(data):
+    res = pd.DataFrame(
+        data.values,
+        index=pd.MultiIndex.from_product(
+            [["h0_0"], data.index], names=["h0", data.index.name]
```

Comment:
Is it true, that this strings are always named like that? I think they can also named differently. So I assume that you would pass the name to this function and use them here. 

Please check this.

Reply:
The name could be anything as it will be droped in the end. Yeah, I will change it to a parameter.